node/nodeTransactions.go

```diff
@@ -221,5 +205,36 @@ func (n *Node) prepareUnsignedTx(tx *smartContractResult.SmartContractResult) (*
 		Data:      string(tx.GetData()),
 		Code:      string(tx.GetCode()),
 		Signature: "",
+		Status:    status,
 	}, nil
 }
+
+func (n *Node) computeTransactionStatus(tx data.TransactionHandler, isInPool bool) string {
+	selfShardID := n.shardCoordinator.SelfId()
+	receiverShardID := n.shardCoordinator.ComputeId(tx.GetRcvAddr())
+
+	var senderShardID uint32
+	if sndAddr := tx.GetSndAddr(); sndAddr != nil {
```

Comment:
Use a "standalone" variable for `sndAddress` instead of `if, assignment, check`. Easier to read.

Reply:
done.